### PR TITLE
MCKIN-12641 - GW submission temporary S3 link

### PR DESCRIPTION
**Pull request by naeem91**
Temporary S3 link valid for 30 mintues is generated for GW submission URL

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @naeem91! I've created [OSPR-3991](https://openedx.atlassian.net/browse/OSPR-3991) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

File: edx_solutions_projects/utils.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        signed_url = s3_client.generate_presigned_url(
+            ClientMethod='get_object',
+            ExpiresIn=S3_FILE_URL_TIMEOUT,
+            Params={
+                'Bucket': settings.AWS_STORAGE_BUCKET_NAME,
+                'Key': file_path
+            }
+        )
+        return signed_url
+    else:
```
Comment (nasirhjafri) [inline edx_solutions_projects/utils.py]: No need for else.

File: edx_solutions_projects/utils.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import boto3
+
+from django.conf import settings
+
+
+S3_FILE_URL_TIMEOUT = 60 * 30  # 30 minutes
```
Comment (nasirhjafri) [inline edx_solutions_projects/utils.py:6]: Would be better if we could pick up from settings, or something like that.
`settings.get('S3_FILE_URL_TIMEOUT', DEFAULT_S3_FILE_URL_TIMEOUT)`

**Review by nasirhjafri — CHANGES_REQUESTED** [review]

**Review by mudassir-hafeez — APPROVED** [review]
LGTM :+1:

File: edx_solutions_projects/utils.py (hunk 2 above)
Comment (naeem91) [reply edx_solutions_projects/utils.py:6]: Actually this is very specific to GW submission files that's why we didn't put it in settings

**Pull request merged**